src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1516,6 +1525,291 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
         raise CLIInternalError("Exception caught while running troubleshoot: {}".format(str(ex)))
 
 
+def display_diagnostics_report(temp_kubeconfig_path, kube_config):   # pylint: disable=too-many-statements
+    if not which('kubectl'):
+        raise CLIError('Can not find kubectl executable in PATH')
+    subprocess_cmd = ["kubectl", "get", "node", "--no-headers"]
```

Comment:
create a kubectl_priori based on kubeconfig and kubecontext passed to the command. That is, if either of them is passed, kubectl_priori = [ "kubectl", "--kubeconfig"......] , so that later in code, you can just append this list to the command and get done with.